core/clustersmngr/factory_caches.go

```diff
@@ -9,18 +9,53 @@ import (
 	"github.com/cheshir/ttlcache"
 	"github.com/weaveworks/weave-gitops/pkg/server/auth"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
 type Clusters struct {
 	sync.RWMutex
 	clusters []Cluster
+	names    map[string]Cluster
```

Comment:
Does this name contain a namespace? Or would this become funny if both the `data-science` and the `frontend` namespaces defined a cluster `dev`?

Reply:
It's only the name of the cluster, since Cluster object doesn't include a namespace 🤔 but it contains the name and server, could we make use of them for uniqueness?